src/kagglehub/models.py

```diff
@@ -18,5 +25,27 @@ def model_download(handle: str, path: Optional[str] = None):
     return registry.resolver(h, path)
 
 
-def model_upload():
-    raise NotImplementedError()
+def model_upload(handle: str, local_model_dir: str, license_name: str, model_type: str, version_notes: Optional[str] = None):
```

Comment:
`model_upload` should not take `model_type`. It can only upload models. It should always pass the model type when calling `upload_files`. My previous comment was about making `upload_files` generic not, this `model_upload`.

We will eventually add a `dataset_upload` which will call `upload_files(local_model_dir, "dataset")

Reply:
so passing in something like "model" should be good, right?